### ISSUE TITLE
verify different validator join/leave scenarios

### ISSUE THREAD

**Issue by troggy** [leapdao/leap-node#368]
# Bounty

Prerequisite #364

Before onboarding validators on a testnet we want to be sure that period submission will work in different situations.

## Scope

For all the scenarios verify at least blocks production and period submissions. E.g. use local environment (leapdao/leap-sandbox)
- scenario 1: validator added to the end of the slot list
- scenario 2: validator removed from the middle or start of the list. In this case we have a gap in validator slots. We should account for that in both contracts and the node code — e.g.  don't rely on `epochLength` as a number of validators anymore. Check submissions before and after slot activation.
- scenario 3: validator added in the middle of the slot list
- scenarios 4: one or many (but less then quorum) validator nodes were restarted
- put down all the issues found into Github issues
- fixing issues found is out of scope of this bounty.

## Deliverables

- (probably) updated github issues list

## Gain for the project
more confidence for testnet commander

## Roles
bounty gardener: @troggy / 40 DAI
bounty worker: name / share
bounty reviewer: name / share

## Funded by

Dev Circle

**Comment by troggy** [comment]
- [ ] scenario 1: validator added to the end of the slot list
- [ ] scenario 2: validator removed from the middle or start of the list. In this case we have a gap in validator slots. We should account for that in both contracts and the node code — e.g. don't rely on epochLength as a number of validators anymore. Check submissions before and after slot activation.
- [ ] scenario 3: validator added in the middle of the slot list
- [ ] scenarios 4: one or many (but less then quorum) validator nodes were restarted

**Comment by troggy** [comment]
Issues:
- https://github.com/leapdao/leap-node/issues/382

**Comment by troggy** [comment]
stopped the work here. Blocked by leapdao/leap-contracts#263 and #382

**Comment by troggy** [comment]
scenario 2 requires #418 to pass
